### PR TITLE
[4/N] Remove unused functions 

### PR DESCRIPTION
**Pull request by cyyever**
Follows #128179

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: aten/src/ATen/native/LossNLL2d.cpp [hunk 1]
```diff
@@ -499,13 +499,4 @@ Tensor nll_loss2d_symint(const Tensor & self, const Tensor & target, const std::
   return std::get<0>(at::nll_loss2d_forward_symint(self, target, weight, reduction, std::move(ignore_index)));
 }
 
-// Duplicate of above code for non-symbolic ints. Kept for BC purposes and to minimize breakages.
```
Comment (Skylion007) [inline aten/src/ATen/native/LossNLL2d.cpp:502]: This says kept for BC purposes, will this really not break anything?

File: aten/src/ATen/native/NamedTensor.cpp [hunk 2]
```diff
@@ -372,21 +366,12 @@ Tensor index_select(const Tensor& self, Dimname dim, const Tensor& index) {
 Tensor scatter(const Tensor& self, Dimname dim, const Tensor& index, const Tensor& source) {
   reportNYIDimnameOverload("scatter");
 }
-static Tensor& scatter_(Tensor& self, Dimname dim, const Tensor& index, const Tensor& source) {
-  reportNYIDimnameOverload("scatter");
-}
 Tensor scatter(const Tensor& self, Dimname dim, const Tensor& index, const Scalar& source) {
   reportNYIDimnameOverload("scatter");
 }
-static Tensor& scatter_(Tensor& self, Dimname dim, const Tensor& index, const Scalar& source) {
```
Comment (Skylion007) [inline aten/src/ATen/native/NamedTensor.cpp:381]: These are stub method... Maybe the real bug here is that they shouldn't be static?

File: aten/src/ATen/native/TensorCompare.cpp [hunk 3]
```diff
@@ -792,12 +792,6 @@ std::tuple<Tensor, Tensor> max(const Tensor& self, Dimname dim, bool keepdim) {
 std::tuple<Tensor&, Tensor&> max_out(const Tensor& self, Dimname dim, bool keepdim, Tensor& max, Tensor& max_indices) {
   return at::max_out(max, max_indices, self, dimname_to_position(self, dim), keepdim);
 }
-static Tensor argmax(const Tensor& /*self*/, Dimname /*dim*/, bool /*keepdim*/) {
```
Comment (Skylion007) [inline aten/src/ATen/native/TensorCompare.cpp:795]: Also, it doesn't make sense to remove stub methods.

File: aten/src/ATen/native/NamedTensor.cpp (hunk 2 above)
Comment (cyyever) [reply aten/src/ATen/native/NamedTensor.cpp:381]: They have been static for a long time.

File: aten/src/ATen/native/LossNLL2d.cpp (hunk 1 above)
Comment (cyyever) [reply aten/src/ATen/native/LossNLL2d.cpp:502]: This file is rarely changed. Maybe the comments were added long time ago.

File: aten/src/ATen/native/TensorCompare.cpp (hunk 3 above)
Comment (cyyever) [reply aten/src/ATen/native/TensorCompare.cpp:795]: Let's try it and see the effects.

**Pull request closed without merging**